packages/core/saved-objects/core-saved-objects-migration-server-internal/src/zdt/utils/get_current_index.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { escapeRegExp } from 'lodash';
+import type { FetchIndexResponse } from '../../actions';
+
+export const getCurrentIndex = (
+  indices: FetchIndexResponse,
+  indexPrefix: string
+): string | undefined => {
+  const matcher = new RegExp(`^${escapeRegExp(indexPrefix)}[_](?<counter>\\d+)$`);
+
+  let lastCount = -1;
+  Object.keys(indices).forEach((indexName) => {
+    const match = matcher.exec(indexName);
+    if (match && match.groups?.counter) {
+      const suffix = parseInt(match.groups.counter, 10);
+      lastCount = Math.max(lastCount, suffix);
+    }
+  });
```

Comment:
Note: there should *always* be a single `[prefix]_1` index present, but just in case of (and because I wanted to add a named group regexp), we're scanning all the indices for the highest suffix.

Reply:
NIT if somebody entered `'.kibana_03'` the function would return `'.kibana_3'`.